Azure.ResourceManager.Core/Resources/LocationContainer.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.ResourceManager.Core.Adapters;
+using Azure.ResourceManager.Resources;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// Represents an Azure geography region where supported resource providers live.
+    /// </summary>
+    public class LocationContainer : OperationsBase
+    {
+        public LocationContainer(AzureResourceManagerClientOptions clientOptions, SubscriptionOperations subscriptionOperations)
+            : base(clientOptions, subscriptionOperations.Id)
+        {
+        }
+
+        protected override ResourceType ValidResourceType => SubscriptionOperations.ResourceType;
+
+        /// <summary>
+        /// Gets the resource client.
+        /// </summary>
+        internal ResourcesManagementClient GetResourcesClient(string subscription) => ClientOptions.GetClient((uri, credential) => new ResourcesManagementClient(uri, subscription, credential));
+
+        /// <summary>
+        /// Gets the subscription client.
+        /// </summary>
+        internal SubscriptionsOperations SubscriptionsClient => GetResourcesClient(Guid.NewGuid().ToString()).Subscriptions;
+
+        /// <summary>
+        /// Gets the Azure subscriptions.
+        /// </summary>
+        /// <returns> Subscription container. </returns>
+        public SubscriptionContainer GetSubscriptionContainer()
+        {
+            return new SubscriptionContainer(ClientOptions);
+        }
+
+        /// <summary>
+        /// Gets the default Azure subscription.
+        /// </summary>
+        public SubscriptionOperations DefaultSubscription { get; private set; }
```

Comment:
we should not be duplicating this from armclient class

You can store the subscription the container was created from in the constructor and have access to that as a private field.

Reply:
remove this